src/Cis194/Hw/Week1.hs

```diff
@@ -27,7 +34,16 @@ type Peg = String
 type Move = (Peg, Peg)
 
 hanoi :: Integer -> Peg -> Peg -> Peg -> [Move]
-hanoi _ _ _ _ = []
+hanoi 1 a b _ = [(a, b)]
+hanoi 0 _ _ _ = []
+hanoi n a b c = (hanoi (n - 1) a c b) ++ (hanoi 1 a b c) ++ (hanoi (n - 1) c b a)
 
 hanoi4 :: Integer -> Peg -> Peg -> Peg -> Peg -> [Move]
-hanoi4 _ _ _ _ _ = []
+hanoi4 2 a b c _ = [(a, c), (a, b), (c, b)]
+hanoi4 1 a b _ _ = [(a, b)]
+hanoi4 0 _ _ _ _ = []
+hanoi4 n a b c d =
+  (hanoi4 (n - m) a c b d) ++
+  (hanoi m a b d) ++
+  (hanoi4 (n - m) c b a d)
+  where m = floor (sqrt (fromIntegral (n * 2)))
```

Comment:
You found a minimal solution! That is awesome. I looked at it for quite a while and gave up. I also love that you used hanoi in this solution.

Reply:
@adkron I found out that it's not actually an optimal solution (technically no solution has been _proven_ to be optimal, but many believe that [this](https://en.wikipedia.org/wiki/Tower_of_Hanoi#Frame.E2.80.93Stewart_algorithm) is optimal). I didn't properly form a recurrence relation when analyzing the algorithm...mostly because it was a complex case and my math skills are weak. However, I got close enough to satisfy the specs, which made me happy.